packages/demos/stories/DragDrop.tsx

```diff
@@ -0,0 +1,135 @@
+/**
+ * © 2020 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayList from '@clayui/list';
+import {XYCoord} from 'dnd-core';
+import update from 'immutability-helper';
+import React, {useState, useCallback, useRef} from 'react';
+import {useDrag, useDrop, DropTargetMonitor} from 'react-dnd';
+
+export interface IDraggableListItemProps {
+	/**
+	 * The text displayed in the List Item
+	 */
+	text: string;
+	/**
+	 * Index of the List Item in order to keep track of it's positioning within the array
+	 */
+	index: number;
+	/**
+	 * Function that handles the moving of List Items
+	 */
+	moveListItem: (dragIndex: number, hoverIndex: number) => void;
+}
+
+interface IDragItem {
+	index: number;
+	type: string;
+}
+
+const DraggableListItem: React.FC<IDraggableListItemProps> = ({
+	index,
+	moveListItem,
+	text,
+}) => {
+	const ref = useRef<HTMLDivElement>(null);
+
+	const [, drop] = useDrop({
+		accept: 'listItem',
+		hover(item: IDragItem, monitor: DropTargetMonitor) {
+			if (!ref.current) {
+				return;
+			}
+
+			const dragIndex = item.index;
```

Comment:
you could also set a default name in the above arg as `{index: dragIndex}` to avoid having to reset it.

Reply:
Hey @kresimir-coko could you make this change as well?

Reply:
I don't understand what exactly you're trying to say here.
```
const dragIndex = item.index;
const hoverIndex = index;
```
Even if you got confused and wanted me to do `{index: hoverIndex}` I'm unsure where in the code to make the change. 
```
const DraggableListItem: React.FunctionComponent<IDraggableListItemProps> = ({
	id,
	index: hoverIndex,
	onMove,
	text,
}) => {
```
Adding `index: hoverIndex˙to the list of params throws an error:
`ERROR in [at-loader] ./packages/demos/stories/DragDrop.tsx:105:14 
    TS18004: No value exists in scope for the shorthand property 'index'. Either declare one or provide an initializer.`

Reply:
Oh I was thinking of moving it to the functions arguments. at `hover(item: IDragItem, monitor: DropTargetMonitor) {`

But I realized you set the index value at the end of the function so its probably best to leave it as is for now.